yaas/objects.py

```diff
@@ -0,0 +1,140 @@
+# coding: utf-8
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+import collections
+
+class Configuration(collections.defaultdict):
+    def __init__(self):
+        super(Configuration, self).__init__(dict)
+
+    @classmethod
+    def parse(cls, raw):
+        ret = cls()
+        for config_block in raw:
+            for name, configs in config_block.items():
+                ret[name].update(configs)
+        return ret
+
+    def serialize(self):
+        ret = []
+        for name, configs in self.items():
+            ret.append({name: configs})
+        return ret
+
+class HostGroup(object):
+    def __init__(self, name, components, cardinality=None, configuration=None):
+        self.name = name
+        self.components = components
+        self.cardinality = cardinality
+        self.configuration = configuration
+
+    @classmethod
+    def parse(cls, raw):
+        fields = {
+            'name': raw['name'],
+            'components': [obj['name'] for obj in raw['components']],
+        }
+
+        if 'cardinality' in raw:
+            fields['cardinality'] = int(raw['cardinality'])
+
+        if 'configurations' in raw:
+            fields['configuration'] = Configuration.parse(raw['configurations'])
+
+        return cls(**fields)
+
+    def serialize(self):
+        components = [{'name': c} for c in self.components]
+        ret = {"name": self.name, "components": components}
+
+        if self.configuration is not None:
+            ret['configurations'] = self.configuration.serialize()
+
+        if self.cardinality is not None:
+            ret['cardinality'] = self.cardinality
+
+        return ret
+
+class Security(object):
+    def __init__(self, security_type, kerberos_descriptor=None, kerberos_descriptor_reference=None):
+        self.type = security_type
+        self.kerberos_descriptor = kerberos_descriptor
+        self.kerberos_descriptor_reference = kerberos_descriptor_reference
+
+    @classmethod
+    def parse(cls, raw):
+        fields = {
+            'security_type': raw['type'],
+            'kerberos_descriptor': raw.get('kerberos_descriptor'),
+            'kerberos_descriptor_reference': raw.get('kerberos_descriptor_reference'),
+        }
+
+        return cls(**fields)
+
+    def serialize(self):
+        ret = {"type": self.type}
+
+        if self.kerberos_descriptor is not None:
+            ret['kerberos_descriptor'] = self.kerberos_descriptor
+
+        if self.kerberos_descriptor_reference is not None:
+            ret['kerberos_descriptor_reference'] = self.kerberos_descriptor_reference
+
+        return ret
+
+class Blueprint(object):
+    # pylint: disable=too-many-arguments
```

Comment:
Maybe put this at the end of 95? Otherwise it will disable this check from here down I think. (Was that your intent?)